babykicks/src/pages/count.js

```diff
@@ -34,28 +34,32 @@ function Count() {
 
   const handleSessionClick = async () => {
     if (currentUser) {
-        const docRef = doc(db, "user", currentProfile.email);
-        const docSnap = await getDoc(docRef);
-        if(docSnap.exists()){
-          console.log("document history data",docSnap.data().history);
-          const updatedProfileData={
-            history:docSnap.data().history
-          }
-          setCurrentProfile(updatedProfileData)
-          
-        }else{
-          console.log("no such document!")
-        }
-      
-      const start = Date.now();
-      // const docRef = doc(db, "user", currentProfile.email);
-      console.log(`********`)
-      console.log(currentProfile)
-      console.log(`********`)
+      console.log("handlesession current profile", currentProfile);
+      console.log("handlesession currentUser", currentUser);
+      const docRef = doc(db, "user", currentProfile.email);
+      const docSnap = await getDoc(docRef);
+      if (docSnap.exists()) {
+        console.log("document history data", docSnap.data().history);
+        const updatedProfileData = {
+          ...currentProfile,
```

Comment:
Fixed the error of currentProfile being overwritten by the history array which had removed the user email address.